File: Service/Implement/ATTPInfoProductGroupsService.cs
using Service.Interface;

namespace Service.Implement
{
	public class ATTPInfoProductGroupsService : BaseService<ATTPInfoProductGroups, IATTPInfoProductGroupsRepository>
	, IATTPInfoProductGroupsService
	{
		private readonly IATTPInfoProductGroupsRepository _ATTPInfoProductGroupsRepository;

		private readonly IProductGroupService _ProductGroupService;
		public ATTPInfoProductGroupsService(IATTPInfoProductGroupsRepository ATTPInfoProductGroupsRepository
			
			, IProductGroupService productGroupService
			
			) : base(ATTPInfoProductGroupsRepository)
		{
			_ATTPInfoProductGroupsRepository = ATTPInfoProductGroupsRepository;
			_ProductGroupService = productGroupService;
		}

		public override void Initialization(ATTPInfoProductGroups model)
		{			
			if (model.ProductGroupID > 0)
			{
				model.Name = _ProductGroupService.GetByID(model.ProductGroupID.Value).Name;
			}			
		}
	}
}

File: CMS/src/app/shared/ATTPInfoProductGroups.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { ATTPInfoProductGroups } from 'src/app/shared/ATTPInfoProductGroups.model';
import { BaseService } from './Base.service';
@Injectable({
  providedIn: 'root'
})
export class ATTPInfoProductGroupsService extends BaseService {

  List: ATTPInfoProductGroups[] | undefined;
  ListFilter: ATTPInfoProductGroups[] | undefined;
  FormData!: ATTPInfoProductGroups;

  DisplayColumns001: string[] = ['ProductGroupID', 'Save'];
  DisplayColumns002: string[] = ['ProductGroupID'];

  constructor(public httpClient: HttpClient) {
    super(httpClient);
    this.Controller = "ATTPInfoProductGroups";
  }
}

